extensions/wikia/Sitemap/SpecialSitemap_body.php

```diff
@@ -477,4 +491,66 @@ public static function onLoadExtensionSchemaUpdates( $updater = null ) {
 
 		return true;
 	}
+
+	/**
+	 * Query Solr (once) to find out which NS_FILE pages are indexed in Solr
+	 *
+	 * Later we use this information to only expect a snippet for pages that are indexed in Solr.
+	 * This means no querying Solr for information it doesn't have and no falling back to parsing
+	 * wikitext for mostly empty pages (if they weren't empty, they would be indexed).
+	 *
+	 * If more there are more than self::SOLR_LIMIT documents, we'll always return true.
+	 *
+	 * @param $pageId
+	 * @return bool
+	 */
+	private function hasSnippet( $pageId ) {
+		global $wgCityId;
+
+		static $pageIdsWithSnippets = null;
+		static $alwaysReturnTrue = false;
+
+		if ( is_null( $pageIdsWithSnippets ) ) {
+			$pageIdsWithSnippets = [];
+
+			$luceneQuery = sprintf( 'wid:%d AND is_image:false AND ns:%d', $wgCityId, NS_FILE );
+
+			$config = new Wikia\Search\Config();
+			$config->setDirectLuceneQuery( true );
+			$config->setLimit( 1 );
+			$config->setQuery( $luceneQuery );
+
+			$queryServiceFactory = new Wikia\Search\QueryService\Factory();
+			$response = $queryServiceFactory->getFromConfig( $config )->search();
+
+			$numResults = $response->getResultsFound();
+
+			if ( $numResults > self::SOLR_LIMIT ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)